polars/polars-utils/src/wasm.rs

```diff
@@ -5,6 +5,14 @@ impl Pool {
         rayon::current_num_threads()
     }
 
+    pub fn current_thread_index(&self) -> Option<usize> {
```

Comment:
Why was this added?

Reply:
it's used in [`polars-core/src/frame/groupby/hashing.rs`](https://github.com/pola-rs/polars/blob/main/polars/polars-core/src/frame/groupby/hashing.rs#L29) and [`polars-core/src/series/implementations/struct_.rs`](https://github.com/pola-rs/polars/blob/main/polars/polars-core/src/series/implementations/struct_.rs#L283).

For added context, [`crate::POOL`](https://github.com/pola-rs/polars/blob/main/polars/polars-core/src/lib.rs#L65) is configured to a custom struct when building on wasm rather than the corresponding lazily-built `rayon::ThreadPool`.